edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/EdgeHubDesiredProperties.cs

```diff
@@ -6,18 +6,19 @@ namespace Microsoft.Azure.Devices.Edge.Hub.Core.Config
 
     public class EdgeHubDesiredProperties
     {
-        [JsonConstructor]
-        public EdgeHubDesiredProperties(string schemaVersion, IDictionary<string, string> routes, StoreAndForwardConfiguration storeAndForwardConfiguration)
-        {
-            this.SchemaVersion = schemaVersion;
-            this.Routes = routes;
-            this.StoreAndForwardConfiguration = storeAndForwardConfiguration;
-        }
-
-        public string SchemaVersion { get; }
-
-        public IDictionary<string, string> Routes { get; }
-
+        [JsonConstructor]
+        public EdgeHubDesiredProperties(string schemaVersion, IDictionary<string, RouteConfiguration> routes, StoreAndForwardConfiguration storeAndForwardConfiguration)
+        {
+            this.SchemaVersion = schemaVersion;
+            this.Routes = routes;
+            this.StoreAndForwardConfiguration = storeAndForwardConfiguration;
+        }
+
+        public string SchemaVersion { get; }
+
+        [JsonConverter(typeof(RouteConfigurationDictionaryConverter))]
+        public IDictionary<string, RouteConfiguration> Routes;

```

Comment:
Just want to bring this existing issue: IDctionary doesn't guarantee the order; should we use OrderedDictionary https://docs.microsoft.com/en-us/dotnet/api/system.collections.specialized.ordereddictionary?redirectedfrom=MSDN&view=netframework-4.8.

Reply:
list is actually a better approach now that we have a customer converter.  I don't see any place where we're actually using this as a dictionary.

---
In reply to: [376650794](https://github.com/Azure/iotedge/pull/2469#discussion_r376650794) [](ancestors = 376650794)

Reply:
Sure.  Agreed this existing issue, please create a bug for tracking.  thanks.

Reply:
6257467

---
In reply to: [377442952](https://github.com/Azure/iotedge/pull/2469#discussion_r377442952) [](ancestors = 377442952)